roofit/roofitcore/src/RooPlot.cxx

```diff
@@ -213,9 +212,8 @@ RooPlot::RooPlot(const RooAbsRealLValue &var, double xmin, double xmax, Int_t nb
   _hist->GetSumw2()->Set(0) ;
   _hist->SetDirectory(nullptr);
 
-  // plotVar can be a composite in case of a RooDataSet::plot, need deepClone
-  _plotVarSet = (RooArgSet*) RooArgSet(var).snapshot() ;
-  _plotVarClone= (RooAbsRealLValue*)_plotVarSet->find(var.GetName()) ;
+  // In the past, the plot variable was cloned, but there was no apparent reason for doing so.
```

Comment:
Are we sure that there are not cases when running without batch mode where we do need to clone ?

Reply:
According to a comment, it's because the "plotVar can be a composite in case of a RooDataSet::plot". Probably
with "composite" it means "non fundamental".

I can't follow hat argument though, because even for derived
variables in a RooDataSet, there is a corresponding fundamental
`RooAbsReal` that is synced with the derived variable.

I guess, since the comment is very old, it was necessary to do the
cloning in an older state of RooFit, but not anymore. It would be difficult to figure out the real reason, maybe Wouter knows though.

In case it turns out from user feedback that the cloning is really
necessary in some scenarios, we can still bring it back. Maybe in an
if-statement that covers these specific cases.

In any case, the benefits of not having to deal with the clones variables outweigh the possible user support burden of fixing these corner cases after they are reported.